app/javascript/components/model/Result.js

```diff
@@ -1,22 +1,26 @@
 import { observable, decorate} from "mobx"
 
 export default class Result {
-  totalItems = null // Int
-  currentPage = 1 // Int
-  itemsPerPage = null // Int
-  records = [] // [String:Any]
 
   constructor(json) {
-    this.totalItems = json["total_count"]
-    this.currentPage = json["page_num"];
-    this.itemsPerPage = json["page_size"]
-    this.records.length = 0;
 
-    let records = json["records"]
-    for (let i=0; i<records.length; i++) {
-      records[i].no = (this.itemsPerPage*(this.currentPage-1))+(i+1)
+    this.totalItems = null // Int
+    this.currentPage = 1 // Int
+    this.itemsPerPage = null // Int
+    this.records = [] // [String:Any]
+
+    if(json) {
```

Comment:
Why did you add a "if (json)" this is different than the original logic?   should this be removed?

Reply:
seems like a reasonable safeguard against an impending undefined.length in the for loop at the least. better solution might be to leave it and add a console.warn or similar in the `else` 
But as I'm not a react developer I will just remove it. I'm  not sure why my username is attached to these react changes, must have been from when I recreated the 6.0.0 state for review.